File: Code blocks/weird_subtraction_process.c
#include<stdio.h>
#include<math.h>
int main(){
    long long int a,b;
    scanf("%lld%lld",&a,&b);
    while(a!=0&&b!=0){
        if(a>=(b+b)){
            a=a-floor(a/b)*b;
        }
        else if(b>=(a+a)){
            b=b-floor(b/a)*a;
        }
        else{
            break;
        }
    }
    printf("%lld %lld",a,b);
    return 0;
}

File: Python/weird_subtraction_process.py
import math

a,b=list(map(int,input().split()))
while a!=0 and b!=0:
    if a>=(2*b):
        c=math.floor(a/b)
        while c>0:
            a=a-b
            c-=1
        print(a,b)
    elif b>=(2*a):
        b=b-(math.floor(b/a))*a
    else:
        break
    break
        
